Upload answered quests

errolpais opened the pull request:
Hi @westnordost, 

This addresses issue #75. I added a new upload icon into the app bar. I tested it manually prior to the java 8 upgrade and it looked good. I just re based my commits for this pull request. The app deploys and looks alright on my phone, but will test it out manually tomorrow.

I wanted to get some feedback on the pull request. I did pull out the ems configuration to make things fit and it seems alright but I might be missing something, Please let me know what you think and if you'd like me to make any additions/changes.

Thanks

westnordost commented on app/src/main/java/de/westnordost/streetcomplete/statistics/UploadsCounter.java:
```diff
@@ -0,0 +1,53 @@
+import de.westnordost.streetcomplete.data.QuestStatus;
+import de.westnordost.streetcomplete.data.osm.persist.OsmQuestDao;
+
+public class UploadsCounter extends android.support.v7.widget.AppCompatTextView {
```
The name for the new class is confusing. After all, this is the counter of answers that have _not_ been uploaded yet.

The other (original) counter counts the number of quest answers that have been uploaded already. This one counts the number of newly added quest answers that are pending to be uploaded. The class and field names should reflect that.

errolpais replied:
Yeah, I did feel there was something off. How about if I renamed both the classes. UnsyncedAnswersCounter and UploadedAnswersCounter? Both their field names can be answeredQuests since the class name will specify what kind of answered quests they contain.

westnordost replied:
These names would work, yes.
> answeredQuests

I meant the names of the fields in MainActivity.

westnordost commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -33,14 +33,21 @@
                 android:layout_gravity="end"
-                android:minEms="7"
-                android:paddingEnd="8dp"
-                android:paddingStart="8dp"
```
Wouldn't that look very packed on smaller devices? Does it even fit with four-digit counters each?

I think it is not really necessary to display the app's name in the header. This way, there will be more than enough space.

errolpais replied:
Removed the title and moved the counters to the left. I added back the padding. The minEms is adding a lot of space so not sure if its needed. The text being dynamic seems to make enough space for itself and the padding ensures they are not squished. Tested the layout on on a pixel emulator.

westnordost commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -33,14 +33,21 @@
+                android:drawablePadding="2dp"
+                android:drawableStart="@drawable/ic_file_upload_24dp"
+                android:gravity="center_vertical"
+                />
```
The component should have a proper selectable item background because it is clickable.

errolpais replied:
I added 'android:background="?android:attr/selectableItemBackground"' to the component. Assuming that's what you had in mind.

errolpais commented:
* Corrected the indentation in the lambdas.
* Renamed counter classes to UnsyncedAnswersCounter and UploadedAnswersCounter
* Set background property for UnsyncedAnswersCounter to selectableItemBackground since it a button
* Added Padding back
* Removed Title and moved Counters to left side of the toolbar

errolpais commented:
Yay! Thanks for the feedback and for maintaining the app 👍
